### PR TITLE
Subscribe / Unsubscribe functionality

### PR DESCRIPTION
**Pull request by gcapalbo**
Adds the ability to define opt-in and opt-out keywords per gateway type.  When a phone number texts in an opt-out keyword, that phone number is added to a list and no more sms are sent to that number until it opts back in by texting in one of the opt-in keywords.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
@gcapalbo the build fail looks real and introduced by the new south dependency on the sms app. I've seen this before so will take a quick crack at it.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
https://github.com/dimagi/commcare-hq/commit/b0ac5c20045a8a42e066d001e8dc4429ca40dc80

File: corehq/apps/sms/models.py [hunk 1]
```diff
@@ -427,3 +443,75 @@ def case_changed_receiver(sender, case, **kwargs):
...
 case_post_save.connect(case_changed_receiver, CommCareCase)
 
+
+class PhoneNumber(models.Model):
+    """
+    Represents a single phone number. This is not intended to be a
+    comprehensive list of phone numbers in the system (yet). For
+    now, it's only used to prevent sending SMS/IVR to phone numbers who
+    have opted out.
+    """
```
Comment (czue) [inline corehq/apps/sms/models.py]: we still only support one verified user - phone number mapping right? would it make sense to store that here for future ease of lookups?

**Comment by czue** [comment]
reviewed quickly - looks good and didn't see anything major. assume you're satisfied with QA on this?

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/sms/models.py (hunk 1 above)
Comment (gcapalbo) [inline corehq/apps/sms/models.py]: @czue yeah, I think longer term we can move the whole phone registry to this table, but for now just wanted to get it out there to use it for this effort

**Comment by gcapalbo** [comment]
@czue thanks for looking into the build failure, it failed after that on a different issue but that should be fixed now too. I also just added one more commit for a separate thing I noticed.

Regarding QA, yup, this is all set - just waiting for the build to make sure it's good, and then we should be good to merge.

**Pull request merged**